internal/store/runtime_state.go

```diff
@@ -39,16 +39,31 @@ type LocalRuntimeState struct {
 	HasSucceededAtLeastOnce bool
 	PID                     int
 	SpanID                  model.LogSpanID
+
+	readinessProbeSuccessful bool
```

Comment:
Internally within K8s, probes actually return a `Result` (see [kubernetes/pkg/probe/probe.go](https://github.com/kubernetes/kubernetes/blob/a9e2e18cc6737072fef2bce838bcf508a8e708f3/pkg/probe/probe.go#L19-L31)).

Valid results are `Success` / `Warning` / `Failure` / `Unknown`. `Warning` is logically the same as `Success` but means there's some debugging stuff -- it's only used in K8s for HTTP probe when there's an excessive number of redirects triggered. `Unknown` isn't actually used - things w/o probes are set to `Success` immediately and those w/ probes are initialized to `Failure` until probe runs + succeeds.

I think it's reasonable to simplify to a bool in our case. (The actual `Prober` iface can still return a result that gets coerced into a bool.)